addon/mixins/scoped-query-cache.js

```diff
@@ -73,5 +76,21 @@ export default Ember.Mixin.create({
 
       return results;
     });
+  },
+
+  scheduleCacheDecay(delay = CACHE_DECAY_TIMEOUT, callback) {
+    this.decayId = setTimeout(() => {
+      this.cache = this._createCache();
+
+      callback && typeof callback === 'function' && callback();
```

Comment:
So we can have a `clearCache` method instead. That does lead to more flexibility here. I'm open either way.